extensions/typescript-language-features/src/languageFeatures/diagnostics.ts

```diff
@@ -95,8 +95,7 @@ interface LanguageDiagnosticSettings {
 }
 
 function areLanguageDiagnosticSettingsEqual(currentSettings: LanguageDiagnosticSettings, newSettings: LanguageDiagnosticSettings): boolean {
```

Comment:
Instead of inverting the function, how about using `!areLanguageDiagnosticSettingsEqual` where it is called?

Reply:
Sounds like good